icechunk/src/format/mod.rs

```diff
@@ -43,14 +43,19 @@ pub struct ChunkTag;
 #[derive(Hash, Clone, PartialEq, Eq, PartialOrd, Ord)]
 pub struct AttributesTag;
 
+#[derive(Hash, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct NodeTag;
+
 impl private::Sealed for SnapshotTag {}
 impl private::Sealed for ManifestTag {}
 impl private::Sealed for ChunkTag {}
 impl private::Sealed for AttributesTag {}
+impl private::Sealed for NodeTag {}
```

Comment:
what is this 🦭 business?

Reply:
This allows us to use a single class `ObjectId<T>` but have different types for SnapshotIds, ManifestIds, etc. That way you get a compile time error if you pass a ChunkId when an SnapshotId is expected. These Tag classes are "markers" to fill in the `T` in `ObjectId<T>`, it's what makes them different types for different objects. Sealed is so a user cannot create their own Id types by mistake ... for example, `ObjectId<u8>` is not going to work.